freebsd.sh

```diff
@@ -1,14 +1,18 @@
 #!/usr/bin/env sh
 
+set -e
+
 # This bootstraps Puppet on FreeBSD
 
-have_pkg=`grep -sc '^WITH_PKGNG' /etc/make.conf`
+with_pkgng=`grep -sc '^WITH_PKGNG' /etc/make.conf` || true;
```

Comment:
This test can be updated to something more modern.  If `sysctl -n kern.osreldate` is older than `90100` then you can assume `pkg_add` is required, otherwise if newer you can check the return code from `pkg -N`.  FreeBSD 9 is going to be EOL'ed on December 31, 2016 so it may not be worth adding support for `pkg_add`.

https://www.freebsd.org/cgi/man.cgi?query=pkg&apropos=0&sektion=0&manpath=FreeBSD+10.3-RELEASE&arch=default&format=html

https://www.freebsd.org/security/security.html#sup